dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/spi/OpenTelemetryAccessor.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.microprofile.telemetry.spi;
+
+import java.util.Optional;
+
+import com.ibm.ejs.ras.Tr;
+import com.ibm.ejs.ras.TraceComponent;
+import com.ibm.ws.kernel.productinfo.ProductInfo;
+import com.ibm.ws.kernel.service.util.ServiceCaller;
+
+import io.openliberty.microprofile.telemetry.internal.common.info.ErrorOpenTelemetryInfo;
+import io.openliberty.microprofile.telemetry.internal.interfaces.OpenTelemetryInfoFactory;
+
+public class OpenTelemetryAccessor {
+    private static final TraceComponent tc = Tr.register(OpenTelemetryAccessor.class);
+
+    private static final ServiceCaller<OpenTelemetryInfoFactory> openTelemetryInfoFactoryService = new ServiceCaller<OpenTelemetryInfoFactory>(OpenTelemetryAccessor.class, OpenTelemetryInfoFactory.class);
```

Comment:
This effectively makes the bundle the `OpenTelemetryAccessor` class is in the singleton user of the `OpenTelemetryInfoFactory` service from an OSGi service registry perspective.  But the static method `getOpenTelemetryInfo` method leaks objects from the `OpenTelemetryInfoFactory` to other bundles (in user features).

What happens if the `OpenTelemetryInfoFactory` is deactivated and reactivated?  Will these leaked objects still work if a user feature still has access to them?

Reply:
The objects that `OpenTelemetryAccessor` leaks from `OpenTelemetryInfoFactory` to other bundles are objects that are designed to be used by developers inside application code so it is normal for them to escape the OSGi registry.

I've not tested restarting the bundle while an application is running. `OpenTelemetryInfoFactory` itself looks like it will be reasonably robust to that as it stores the relevant objects in ApplicationMetaData so assuming the application itself hasn't been stopped while it was restarting it should be able to pick up where it left off. 

However the objects themselves are created by `OpenTelemetryInfoFactory` calling the upstream OpenTelemetrySdk so I fear those could fall apart if the bundle is restarting and an application or another liberty component tries to use them. This is a pre-existing issue regardless of this SPI, but definitely something to investigate. Thank you for bringing it to my attention.

Reply:
This was discussed in Slack here: https://ibm-cloud.slack.com/archives/C31GEJTQD/p1721399032670279

The summary is that the component itself `OpenTelemetryInfoFactory` is unlikely to restart because the only OSGi service reference it has from outside of our OpenTelemetry feature is the slot service. And any other component that stores `OpenTelemetry` objects will have a dependency on this bundle, so if this bundle gets unistalled they will too